presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/WindowMatcher.java

```diff
@@ -46,19 +72,166 @@ public MatchResult detailMatches(PlanNode node, PlanNodeCost cost, Session sessi
 
         WindowNode windowNode = (WindowNode) node;
 
+        if (!prePartitionedInputs
+                .map(expectedInputs -> expectedInputs.stream()
+                        .map(alias -> alias.toSymbol(symbolAliases))
+                        .collect(toImmutableSet())
+                        .equals(windowNode.getPrePartitionedInputs()))
+                .orElse(true)) {
+            return NO_MATCH;
+        }
+
+        if (!specification
+                .map(expectedSpecification ->
+                        expectedSpecification.getExpectedValue(symbolAliases)
+                        .equals(windowNode.getSpecification()))
+                .orElse(true)) {
+            return NO_MATCH;
+        }
+
+        if (!preSortedOrderPrefix
+                .map(Integer.valueOf(windowNode.getPreSortedOrderPrefix())::equals)
+                .orElse(true)) {
+            return NO_MATCH;
+        }
+
+        if (!hashSymbol
+                .map(expectedHashSymbol -> expectedHashSymbol
+                        .map(alias -> alias.toSymbol(symbolAliases))
+                        .equals(windowNode.getHashSymbol()))
+                .orElse(true)) {
+            return NO_MATCH;
+        }
+
         /*
          * Window functions produce a symbol (the result of the function call) that we might
          * want to bind to an alias so we can reference it further up the tree. As such,
```

Comment:
How about s/As such/Instead of matching window functions here/ to make this more understandable? <- reviewing my own unintelligible comment :-D